Backpack/src/main/java/net/skyscanner/backpack/starrating/BpkStarRating.kt

```diff
@@ -0,0 +1,121 @@
+package net.skyscanner.backpack.starrating
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.annotation.ColorInt
+import androidx.appcompat.widget.LinearLayoutCompat
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.BpkTheme
+import net.skyscanner.backpack.util.BpkTheme.Companion.wrapContextWithDefaults
+import net.skyscanner.backpack.util.createContextThemeWrapper
+
+open class BpkStarRating @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : LinearLayoutCompat(wrapContext(context, attrs), attrs, defStyleAttr) {
+
+  private var _maxRating: Int = 5
+  var maxRating: Int
+    get() = _maxRating
+    set(value) {
+      if (value < 0) {
+        throw IllegalArgumentException("Invalid maxRating=$value")
+      }
+      _maxRating = value
+      update()
+    }
+
+  private var _rating: Float = maxRating / 2f
+  var rating: Float
+    get() = _rating
+    set(value) {
+      _rating = clamp(value, 0f, maxRating.toFloat())
+      update()
+    }
+
+  private var _starColor: Int = BpkTheme.getColor(context, R.color.bpkGray100)
+  var starColor: Int
+    @ColorInt
+    get() = _starColor
+    set(@ColorInt value) {
+      if (_starColor != value) {
+        _starColor = value
+        update()
+      }
+    }
+
+  private var _starFilledColor: Int = BpkTheme.getColor(context, R.color.bpkYellow500)
+  var starFilledColor: Int
+    @ColorInt
+    get() = _starFilledColor
+    set(@ColorInt value) {
+      if (_starFilledColor != value) {
+        _starFilledColor = value
+        update()
+      }
+    }
+
+  init {
+    initialize(attrs, defStyleAttr)
+  }
+
+  private fun initialize(attrs: AttributeSet?, defStyleAttr: Int) {
+    orientation = HORIZONTAL
+    attrs
```

Comment:
This code should run if attrs is null so the themed props are fetched. Don't need to check for null `obtainStyledAttributes` supports passing null as argument

Reply:
Done